test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress001.cpp

```diff
@@ -33,214 +33,221 @@ extern "C" {
+        }
+      }
     }
-    memcpy(javachars->checkstr[index++],digest,DIGESTLENGTH);
-    if (index % printperiod == 0) {
-        printf("Check string sum for thread %.*s is ", elem_len, elem);
-        for (j=0;j<DIGESTLENGTH;j++)
-            printf("%02x", digest[j]);
-        printf("\n");
+    index = 0;
+    len = 0;
+    for (j = 0; j < nstr; j++) {
+      free(javachars->checkstr[j]);
     }
-    free(elem);
-    if (index == nstr) {
-        printf("JNI Unicode strings memory=%ld\n",len);
-        tmpstr=env->NewString(javachars->str[index-1],elem_len); CE
-        for (j=0; j<nstr; j++) {
-            elem = (char*) c_malloc(env, javachars->size[j]*sizeof(char));
-            for (i=0; i < javachars->size[j]; i++) {
-                elem[i] = (char) javachars->str[j][i];
-            }
-            //memcpy(digest, elem, javachars->size[j]);
-            for (i=0;i<DIGESTLENGTH;i++) {
-                digest[i]=0;
-            }
-            for (i=0;i<javachars->size[j]; i++) {
-                digest[i % DIGESTLENGTH]+=elem[i];
-            }
-            free(elem);
-            if (memcmp(digest,javachars->checkstr[j],javachars->size[j]) == 0) {
-                env->ReleaseStringChars(jstr,javachars->str[j]); CE
-                javachars->str[j] = NULL;
-                javachars->checkstr[j] = NULL;
-                javachars->size[j] = 0;
-            }
-            else {
-                equal=0;
-                printf("The Unicode element No. %d has been corrupted\n",j);
-                for (i=0;i<DIGESTLENGTH;i++) {
-                    printf("digest[%d]=%02x checkstr[%d]=%02x\n",i,digest[i],i,javachars->checkstr[j][i]);
-                }
-            }
-        }
-        index=0;
-        len = 0;
-        for (j=0;j<nstr;j++)
-            free(javachars->checkstr[j]);
-        free(javachars->checkstr);
-        free((void *)(javachars->str));
-        free(javachars->size);
-        free(javachars);
-        clazz=env->FindClass(clsName); CE
-        if (!equal) {
-            methodID=env->GetStaticMethodID(clazz, name, sig); CE
-            env->CallStaticVoidMethod(clazz, methodID, JNI_FALSE); CE
-        }
-        //methodID=env->GetStaticMethodID(clazz, halt, haltSig); CE
-        //env->CallStaticVoidMethod(clazz, methodID); CE
-        return(tmpstr);
+    free(javachars->checkstr);
+    free((void *)(javachars->str));
+    free(javachars->size);
+    free(javachars);
+    clazz = env->FindClass(clsName); CE
+    if (!equal) {
+      methodID = env->GetStaticMethodID(clazz, name, sig); CE
+      env->CallStaticVoidMethod(clazz, methodID, JNI_FALSE); CE
     }
-    env->MonitorExit(jobj); CE
-    return(env->NewString(javachars->str[index-1],elem_len));
+    //methodID = env->GetStaticMethodID(clazz, halt, haltSig); CE
+    //env->CallStaticVoidMethod(clazz, methodID); CE
+    return tmpstr;
+  }
+  CHECK(env->MonitorExit(jobj));
+  return env->NewString(javachars->str[index - 1], elem_len);
```

Comment:
Should you check for exceptions after NewString?

Reply:
It isn't essential as any exception will be thrown in the calling Java code. It won't be as fail-fast as using CE but doesn't seem worth the effort to change.